### PR TITLE
WAITP-1429: Autocomplete add and filtering of options

### PR DESCRIPTION
**Pull request by acdunham**
### Changes:
- Enabled autocomplete attribute filtering
- Enabled autocomplete option creation
- Added tests for `AutocompleteQuestion` and `useSubmitSurvey`

### CODE REVIEW

File: packages/datatrak-web/src/api/mutations/useSubmitSurvey.ts [hunk 1]
```diff
@@ -55,35 +82,54 @@ export const processSurveyResponse = ({
...
+        surveyResponse.entity_id = answer as string;
         break;
       }
-      default:
+      case 'Autocomplete': {
+        // if the answer is a new option, add it to the options_created array to be added to the DB
+        const { isNew, value, label, optionSetId } = answer as AutocompleteAnswer;
+        if (isNew) {
+          surveyResponse.options_created.push({
+            option_set_id: optionSetId,
```
Comment (tcaiger) [inline packages/datatrak-web/src/api/mutations/useSubmitSurvey.ts:111]: I noticed that on Meditrak the optionsCreated have an id and a sortOrder, does it need them here too?

**Review by tcaiger — APPROVED** [review]
Looks great. I just left one question just in case.

File: packages/datatrak-web/src/api/mutations/useSubmitSurvey.ts (hunk 1 above)
Comment (acdunham) [reply packages/datatrak-web/src/api/mutations/useSubmitSurvey.ts:111]: I don't think so - it automatically adds the sort order as the the next number (to make the option the last one in the list) and a unique id gets applied when it's created

**Comment by tcaiger** [comment]
Oh yeah I just noticed that the base branch is still pointing at the old one. Ethan made a new one yesterday which is `datatrak-web-epic`

**Pull request merged**